pandas/_libs/indexing.pyi

```diff
@@ -0,0 +1,16 @@
+from typing import (
+    Generic,
+    TypeVar,
+)
+
+from pandas.core.indexing import IndexingMixin
+
+_IndexingMixinT = TypeVar("_IndexingMixinT", bound=IndexingMixin)
```

Comment:
in practice isn't this Series/DataFrame?

Reply:
Yes, that is the case. The few lines that directly call the constructor all call with `IndexMixin`. Since `NDFrameIndexerBase` just saves any provided object, I went with `_IndexingMixinT`.

Here are the few lines that call the constructor:
pandas/core/indexing.py:269
pandas/core/indexing.py:529
pandas/core/indexing.py:578
pandas/core/indexing.py:626

I added a small comment about it most likely being DataFrame/Series.